pkgs/applications/version-management/git-and-tools/git-branchless/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib, fetchFromGitHub
+
+, coreutils
+, git
+, ncurses
+, rustPlatform
+, sqlite
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "git-branchless";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "arxanas";
+    repo = "git-branchless";
+    rev = "v${version}";
+    sha256 = "0pfiyb23ah1h6risrhjr8ky7b1k1f3yfc3z70s92q3czdlrk6k07";
+  };
+
+  cargoSha256 = "0gplx80xhpz8kwry7l4nv4rlj9z02jg0sgb6zy1y3vd9s2j5wals";
+
+  buildInputs = [
+    ncurses
+    sqlite
+  ];
+
+  # Remove path hardcodes patching if they get fixed upstream, see:
+  # https://github.com/arxanas/git-branchless/issues/26
+  postPatch = ''
+    # Inline test hardcodes `echo` location.
+    sed -i -e "s|/bin/echo|${coreutils}/bin/echo|" ./src/commands/wrap.rs
+
+    # Tests in general hardcode `git` location.
+    sed -i -e "s|/usr/bin/git|${git}/bin/git|" ./src/testing.rs
+  '';
+
+  preCheck = ''
+    # Tests require path to git.
+    export PATH_TO_GIT=${git}/bin/git
```

Comment:
That will not work for cross compilation.

Reply:
This one is just for tests -- can you actually run tests when cross compiling?

Or did you mean to refer to the replacements in `postPatch` -- because I think those are actually in the binary.

Do you know how this could be improved?

Reply:
I have tested it, it actually works, building `pkgsCross.raspberryPi.git-branchless`:

```
/nix/store/fqhzq1qviiqwpixwxnw1xihvaw2brnnl-git-branchless-armv6l-unknown-linux-gnueabihf-0.3.2
```

The `bin/git-branchless` contains no references to `bin/echo` or `bin/git`; the former is only used in a `#[test]` section, and the other one `src/testing.rs` might be in testing as well.

It seems that cargo builds already skip running tests when cross compiling:

```
Finished cargoBuildHook
installing
Executing cargoInstallHook
Finished cargoInstallHook
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/fqhzq1qviiqwpixwxnw1xihvaw2brnnl-git-branchless-armv6l-unknown-linux-gnueabihf-0.3.2
shrinking /nix/store/fqhzq1qviiqwpixwxnw1xihvaw2brnnl-git-branchless-armv6l-unknown-linux-gnueabihf-0.3.2/bin/git-branchless
armv6l-unknown-linux-gnueabihf-strip is /nix/store/a4yimbbcm3q1f0ilg5ixig4qhcakbnzm-armv6l-unknown-linux-gnueabihf-binutils-2.35.1/bin/armv6l-unknown-linux-gnueabihf-strip
stripping (with command armv6l-unknown-linux-gnueabihf-strip and flags -S) in /nix/store/fqhzq1qviiqwpixwxnw1xihvaw2brnnl-git-branchless-armv6l-unknown-linux-gnueabihf-0.3.2/bin 
patching script interpreter paths in /nix/store/fqhzq1qviiqwpixwxnw1xihvaw2brnnl-git-branchless-armv6l-unknown-linux-gnueabihf-0.3.2
checking for references to /build/ in /nix/store/fqhzq1qviiqwpixwxnw1xihvaw2brnnl-git-branchless-armv6l-unknown-linux-gnueabihf-0.3.2...
copying 1 paths...
```

So all seems to work with the current approach.